lib/setup/setupInstall.js

```diff
@@ -1586,8 +1592,46 @@ function Install(options) {
                 // find the adapter's io-package.json
                 const adapterNpm = `${tools.appName}.${adapter}`;
                 const ioPack = require(`${adapterNpm}/io-package.json`); // yep, it's that easy
+
                 if (!ioPack.common || !ioPack.common.nondeletable) {
                     await that.npmUninstallAsync(adapterNpm, null, false);
+                    // after uninstalling we have to restart the defined adapters
+                    if (ioPack.common.restartAdapters) {
+                        if (!Array.isArray(ioPack.common.restartAdapters)) {
+                            // its not an array, now it can only be a single adapter as string
+                            if (typeof ioPack.common.restartAdapters !== 'string') return;
+                            ioPack.common.restartAdapters = [ioPack.common.restartAdapters];
+                        }
+                        if (ioPack.common.restartAdapters.length && ioPack.common.restartAdapters[0]) {
+                            const instances = await tools.getAllInstancesAsync(ioPack.common.restartAdapters, objects);
+                            if (instances && instances.length) {
+                                for (const instance of instances) {
+                                    const obj = await getObjectAsync(instance);
+                                    // if instance is enabled
+                                    if (obj && obj.common && obj.common.enabled) {
+                                        obj.common.enabled = false; // disable instance
+
+                                        obj.from = `system.host.${tools.getHostName()}.cli`;
+                                        obj.ts = Date.now();
```

Comment:
Although it is not necessary to have this inside the try block, I would put it there, so it is visually grouped with the corresponding setObjectAsync call.